src/lib/core/OTAImageHeader.h

```diff
@@ -58,6 +58,62 @@ struct OTAImageHeader
     ByteSpan mImageDigest;
 };
 
-CHIP_ERROR DecodeOTAImageHeader(ByteSpan buffer, OTAImageHeader & header);
+class OTAImageHeaderParser
+{
+public:
+    /**
+     * @brief Prepare the parser for accepting Matter OTA image chunks.
+     *
+     * The method can be called many times to reset the parser state.
+     */
+    void Init();
+
+    /**
+     * @brief Clear all resources associated with the parser.
+     */
+    void Clear();
+
+    /**
+     * @brief Returns if the parser is ready to accept subsequent Matter OTA image chunks.
+     */
+    bool IsInitialized() const { return mState != State::kNotInitialized; }
+
+    /**
+     * @brief Decode Matter OTA image header
+     *
+     * The method takes subsequent chunks of the Matter OTA image file and decodes the header when
+     * enough data has been provided.
+     *
+     * @param buffer Byte span containing a subsequent Matter OTA image chunk. When the method
+     *               returns CHIP_NO_ERROR, it contains a remaining part of the chunk, not used
```

Comment:
Do you mean "contains" or "may contain"?

Reply:
I mean contains. It's true each time CHIP_NO_ERROR is returned, though, in particular, the remaining part can be empty.

Reply:
Rephrased this a little bit.